presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -93,6 +94,8 @@
 
     private final IdentityHashMap<SampledRelation, Double> sampleRatios = new IdentityHashMap<>();
 
+    private final IdentityHashMap<QuerySpecification, Symbol> groupIdSymbols = new IdentityHashMap<>();
```

Comment:
I need a way to match the `groupId` symbol that is coming out of the `GroupIdNode` to the corresponding `grouping` function. So I changed the `Analysis` object to keep track of them. My new solution moves that map to the `SymbolAllocator`. Let me know if you have a better solution. See 15c59d02f5eb5c5bf87a8736a36bd03b3ae9323f for the relevant changes.